test/integration/controller_test.go

```diff
@@ -435,26 +495,29 @@ func TestCAPIClusterCreate(t *testing.T) {
 					return
 				}
 				machineSets[i] = ms
+
 				for _, role := range ms.Spec.Template.Spec.Roles {
 					if role == capicommon.MasterRole {
-						if masterMachinSet != nil {
-							t.Fatalf("multiple master machinesets: %s and %s", masterMachinSet.Name, ms.Name)
+						if masterMachineSet != nil {
+							t.Fatalf("multiple master machinesets: %s and %s", masterMachineSet.Name, ms.Name)
 						}
-						masterMachinSet = ms
+						masterMachineSet = ms
 					}
 				}
-				if masterMachinSet == nil {
-					t.Fatalf("no master machineset")
-				}
+				coClusterSpec.MachineSets = append(coClusterSpec.MachineSets, clustopClusterMS)
+			}
+
+			if masterMachineSet == nil {
+				t.Fatalf("no master machineset")
 			}
 
 			if err := waitForCAPIClusterToExist(capiClient, testNamespace, testClusterName); err != nil {
 				t.Fatalf("error waiting for Cluster to exist: %v", err)
 			}
 
-			for _, machineSet := range tc.machineSets {
-				if err := waitForCAPIMachineSetToExist(capiClient, testNamespace, machineSet.Name); err != nil {
-					t.Fatalf("error waiting for MachinSet %v to exist: %v", machineSet.Name, err)
+			for _, clustopClusterMS := range tc.machineSets {
```

Comment:
Iterate over `machineSets` instead so that we can use the full name from the machine set.